backend/app/routes/session_utils/save_trial.py

```diff
@@ -0,0 +1,122 @@
+from datetime import datetime
+from typing import List
+
+from beanie import PydanticObjectId
+
+from models.session import Session
+from models.trial import Trial, Solution, TrialError, WrittenStrategy, Advisor
+from utils.utils import estimate_solution_score
+
+
+async def save_trial(body, session, trial, trial_type):
+    # save trial results
+    if trial_type == 'individual':
+        save_individual_demonstration_trial(trial, body)
+    elif trial_type == 'social_learning_selection':
+        # select all social learning trials for one advisor
+        sl_start = session.current_trial_num
+        sl_end = 1
+        for t in session.trials[sl_start + 1:]:
+            if t.trial_type in ['social_learning_selection', 'individual']:
+                break
+            sl_end += 1
+        trials = session.trials[sl_start:sl_start + sl_end]
+        await save_social_leaning_selection(trials, session.subject_id, body)
+        session.trials[sl_start:sl_start + sl_end] = trials
+    elif trial_type == 'social_learning':
+        save_individual_demonstration_trial(trial, body)
+    elif trial_type == 'demonstration':
+        save_individual_demonstration_trial(trial, body)
+    elif trial_type == 'written_strategy':
+        save_written_strategy(trial, body)
+    elif trial_type in ['consent', 'n_practice', 'debriefing',
+                        'instruction_welcome', 'instruction_learning_selection',
+                        'instruction_learning', 'instruction_learning',
+                        'instruction_individual', 'instruction_demonstration',
+                        'instruction_written_strategy']:
+        save_empty_trial(trial)
+
```

Comment:
I think the trial type can't be unknown here because I check the type and throw an error before https://github.com/vagechirkov/reward-networks-iii-experiment/blob/1b7e2b971e05c2041a5d52138020bf6d279874d2/backend/app/routes/session.py#L52